src/library/dao/analysisdao.cpp

```diff
@@ -323,14 +323,26 @@ void AnalysisDao::saveTrackAnalyses(TrackInfoObject* pTrack) {
     ConstWaveformPointer pWaveform = pTrack->getWaveform();
     ConstWaveformPointer pWaveSummary = pTrack->getWaveformSummary();
 
+    TrackId trackId(pTrack->getId());
+
+    // Delete waveform analysis if track was requested to have its waveform cleared.
+    if (pTrack->isClearWaveformRequested()) {
```

Comment:
The code in both AnalyzerWaveform and AnalysisDao for clearing the waveform if requested looks very similar. I would recommend to encapsulate the code from AnalysisDao in a function:

bool AnalysisDao::clearWaveformIfRequested(TrackInfoObject* pTrack)

Then from AnalyzerWaveform::loadStored() you could simply invoke pAnalysisDao->clearWaveformIfRequested(tio) before obtaining the waveform pointers from the TIO.